addons/account/models/account_account.py

```diff
@@ -346,20 +355,85 @@ def _check_account_is_bank_journal_bank_account(self):
         if self._cr.fetchone():
             raise ValidationError(_("You cannot change the type of an account set as Bank Account on a journal to Receivable or Payable."))
 
+    @api.depends_context('company')
+    def _compute_code(self):
+        values = self.env['ir.property'].with_company(self.env.company.root_id).sudo()._get_multi('code', 'account.account', self.ids)
+        for record in self:
+            # Need to set record.code with `company = self.env.company`, not `self.env.company.root_id`
+            record.code = values.get(record.id)
+
+    def _search_code(self, operator, value):
+        return self._fields['code']._search_company_dependent(self.with_company(self.env.company.root_id), operator, value)
+
+    def _inverse_code(self):
+        values = {
+            # Need to access record.code with `company = self.env.company`
+            record.id: self._fields['code'].convert_to_write(record.code, record)
+            for record in self
+        }
+        self.env['ir.property'].with_company(self.env.company.root_id).sudo()._set_multi('code', 'account.account', values)
+
+    @api.depends_context('company')
     @api.depends('code')
     def _compute_account_root(self):
-        # this computes the first 2 digits of the account.
-        # This field should have been a char, but the aim is to use it in a side panel view with hierarchy, and it's only supported by many2one fields so far.
-        # So instead, we make it a many2one to a psql view with what we need as records.
         for record in self:
-            record.root_id = (ord(record.code[0]) * 1000 + ord(record.code[1:2] or '\x00')) if record.code else False
+            record.root_id = self.env['account.root']._from_account_code(record.code)
+
+    def _search_account_root(self, operator, value):
+        if operator in ['=', 'child_of']:
+            root = self.env['account.root'].browse(value)
+            if operator == '=' and not root.parent_id:
+                return [('id', 'in', [])]
```

Comment:
Using `FALSE_LEAF` would allow to avoid or simplify some queries, but I don't understand the condition